### DISCUSSION TITLE
[Feature] Deinterlace in transcoding

### DISCUSSION THREAD

**Discussion by Goodwu** [immich-app/immich#10562] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

![image](https://github.com/immich-app/immich/assets/348005/9093c779-801c-462e-ad29-7d55b431377f)

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by Goodwu** [comment]
Add "-vf deinterlace_qsv" to the ffmpeg command line can fix this.

**Reply by jamesharding26** [reply]
Could you please elaborate how to do this?

**Reply by Goodwu** [reply]
Now you can only input the command in the console of docker, one file by one file.

**Comment by jamesharding26** [comment]
Hi all, is there anyone that could integrate deinterlacing at a core/top level? There must be a lot of people like me with home videos recorded interlaced from the 80s - 2000s that currently can't be processed properly?

 It would just be an addition to ffmpeg to check for an interlace flag in the file and then Deinterlace the transcoded. 

Thank you!

**Comment by Fishday17** [comment]
I have thousands of mts videos, really need this feature!

**Comment by schulzalxs** [comment]
I’m interested in implementing automatic deinterlacing during video transcoding.

My use case: AVCHD/MTS sources that are H.264 but interlaced (e.g. 1080i, field_order=TFF/BFF). These currently show combing artifacts in web playback and can cause severe playback issues on iOS.

Proposed approach (initial, minimal scope):
- Detect interlaced sources (start with ffprobe field_order; optionally idet later)
- Add a server-side setting like deinterlace: off | auto | force (default off)
- When enabled and interlaced is detected, ensure we re-encode video (avoid stream copy) and apply a deinterlace filter (e.g. bwdif/yadif) in the ffmpeg pipeline

Before I start: is anyone actively working on this already, or are there preferences/constraints on the implementation (filters, default behavior, HW accel considerations)?
Also, is #10562 the right place to coordinate this work, or is there another preferred tracking location?
